core/templates/pages/release-coordinator-page/release-coordinator-page.component.spec.ts

```diff
@@ -0,0 +1,166 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for release coordinator page component.
+ */
+
+import { TestBed, waitForAsync, ComponentFixture, fakeAsync, tick } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { FormBuilder } from '@angular/forms';
+import { PromoBarBackendApiService } from 'services/promo-bar-backend-api.service';
+import { ReleaseCoordinatorBackendApiService } from './services/release-coordinator-backend-api.service';
+import { ReleaseCoordinatorPageConstants } from './release-coordinator-page.constants';
+import { ReleaseCoordinatorPageComponent } from './release-coordinator-page.component';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { PromoBar } from 'domain/promo_bar/promo-bar.model';
+
+describe('Release coordinator page', () => {
+  let component: ReleaseCoordinatorPageComponent;
+  let fixture: ComponentFixture<ReleaseCoordinatorPageComponent>;
+  let pbbas: PromoBarBackendApiService;
+  let rcbas: ReleaseCoordinatorBackendApiService;
+
+  let testPromoBarData = new PromoBar(true, 'Hello');
+  let testMemoryCacheProfile = {
+    peak_allocation: '1430120',
+    total_allocation: '1014112',
+    total_keys_stored: '2'
+  };
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+      ],
+      declarations: [
+        ReleaseCoordinatorPageComponent
+      ],
+      providers: [
+        PromoBarBackendApiService,
+        ReleaseCoordinatorBackendApiService,
+        FormBuilder
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ReleaseCoordinatorPageComponent);
+    component = fixture.componentInstance;
+    pbbas = TestBed.inject(PromoBarBackendApiService);
+    rcbas = TestBed.inject(ReleaseCoordinatorBackendApiService);
+  });
+
+  beforeEach(() => {
+    spyOn(pbbas, 'getPromoBarDataAsync').and.returnValue(
+      Promise.resolve(testPromoBarData));
+    component.ngOnInit();
+  });
+
+  it('should load the component with the correct properties' +
+  'when user navigates to release coordinator page', fakeAsync(() => {
+    expect(component.statusMessage).toEqual('');
+    expect(component.submitButtonDisabled).toBeTrue();
+    expect(component.memoryCacheDataFetched).toBeFalse();
+    expect(component.activeTab).toEqual(
+      ReleaseCoordinatorPageConstants.TAB_ID_BEAM_JOBS);
+
+    tick();
+
+    expect(pbbas.getPromoBarDataAsync).toHaveBeenCalled();
+    expect(component.promoBarConfigForm.enabled).toBeTrue();
+  }));
+
+  it('should update promo bar config and set success status',
+    fakeAsync(() => {
+      spyOn(pbbas, 'updatePromoBarDataAsync').and.returnValue(
+        Promise.resolve());
+
+      component.updatePromoBarConfig();
+      expect(component.statusMessage).toEqual('Updating promo-bar config...');
```

Comment:
```suggestion
      component.updatePromoBarConfig();

      expect(component.statusMessage).toEqual('Updating promo-bar config...');
```

Do the same everywhere else.

Reply:
done